zigup.zig

```diff
@@ -429,6 +448,44 @@ fn listCompilers(allocator: *Allocator) !void {
     }
 }
 
+fn keepOrUnkeepCompiler(allocator: *Allocator, compiler_name_to_keep_or_unkeep: []const u8, keep_or_unkeep: enum { keep, unkeep }) !void {
+    if (mem.eql(u8, "master", compiler_name_to_keep_or_unkeep)) {
+        std.debug.warn("master is always kept, doing nothing\n", .{});
+        return;
+    }
+    const install_dir_string = try getInstallDir(allocator, .{ .create = true });
+    defer allocator.free(install_dir_string);
+
+    // TODO openDirAbsolute in stdlib
+    var install_dir = std.fs.cwd().openDir(install_dir_string, .{ .iterate = true }) catch |e| switch (e) {
+        error.FileNotFound => return,
+        else => return e,
+    };
+    defer install_dir.close();
+
+    var compiler_dir = install_dir.openDir(compiler_name_to_keep_or_unkeep, .{}) catch |e| switch (e) {
```

Comment:
just `compiler_version` is fine.  A name only needs to be as long a necessary to distinguish it from other variables.  Not having a principle like this means we can justify a silly name like this:
 `compiler_name_to_keep_or_unkeep_in_the_keep_function_that_is_a_string_and_is_constant_and_is_useful_because_it_is_a_zig_variable`